<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1500</width>
    <height>1300</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1500</width>
    <height>1200</height>
   </size>
  </property>
  <property name="baseSize">
   <size>
    <width>1500</width>
    <height>1300</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QLineEdit {
	background-color: transparent;
	border: 1px;
	color: black;
	font: bold 19px;
}

QMainWindow{
	background-color: #353538;
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="minimumSize">
    <size>
     <width>1500</width>
     <height>1300</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>900</width>
     <height>900</height>
    </size>
   </property>
   <property name="styleSheet">
    <string notr="true">width:1500px;
height:1300px;
color: white;</string>
   </property>
   <widget class="QDoubleSpinBox" name="moveX">
    <property name="geometry">
     <rect>
      <x>1240</x>
      <y>20</y>
      <width>70</width>
      <height>30</height>
     </rect>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::UpDownArrows</enum>
    </property>
    <property name="keyboardTracking">
     <bool>true</bool>
    </property>
    <property name="minimum">
     <double>-1000.000000000000000</double>
    </property>
    <property name="maximum">
     <double>1000.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="moveY">
    <property name="geometry">
     <rect>
      <x>1330</x>
      <y>20</y>
      <width>70</width>
      <height>30</height>
     </rect>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::UpDownArrows</enum>
    </property>
    <property name="keyboardTracking">
     <bool>true</bool>
    </property>
    <property name="minimum">
     <double>-1000.000000000000000</double>
    </property>
    <property name="maximum">
     <double>1000.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="moveZ">
    <property name="geometry">
     <rect>
      <x>1420</x>
      <y>20</y>
      <width>70</width>
      <height>30</height>
     </rect>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::UpDownArrows</enum>
    </property>
    <property name="keyboardTracking">
     <bool>true</bool>
    </property>
    <property name="minimum">
     <double>-1000.000000000000000</double>
    </property>
    <property name="maximum">
     <double>1000.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="rotateY">
    <property name="geometry">
     <rect>
      <x>1330</x>
      <y>100</y>
      <width>70</width>
      <height>30</height>
     </rect>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::UpDownArrows</enum>
    </property>
    <property name="keyboardTracking">
     <bool>true</bool>
    </property>
    <property name="minimum">
     <double>-360.000000000000000</double>
    </property>
    <property name="maximum">
     <double>360.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="rotateZ">
    <property name="geometry">
     <rect>
      <x>1420</x>
      <y>100</y>
      <width>70</width>
      <height>30</height>
     </rect>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::UpDownArrows</enum>
    </property>
    <property name="keyboardTracking">
     <bool>true</bool>
    </property>
    <property name="minimum">
     <double>-360.000000000000000</double>
    </property>
    <property name="maximum">
     <double>360.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="rotateX">
    <property name="geometry">
     <rect>
      <x>1240</x>
      <y>100</y>
      <width>70</width>
      <height>30</height>
     </rect>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::UpDownArrows</enum>
    </property>
    <property name="keyboardTracking">
     <bool>true</bool>
    </property>
    <property name="minimum">
     <double>-360.000000000000000</double>
    </property>
    <property name="maximum">
     <double>360.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="scaleXYZ">
    <property name="geometry">
     <rect>
      <x>1360</x>
      <y>190</y>
      <width>70</width>
      <height>30</height>
     </rect>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::UpDownArrows</enum>
    </property>
    <property name="keyboardTracking">
     <bool>true</bool>
    </property>
    <property name="decimals">
     <number>2</number>
    </property>
    <property name="minimum">
     <double>0.010000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
    <property name="value">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QPushButton" name="file">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>1230</y>
      <width>131</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>choose file</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="file_inf">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>1220</y>
      <width>281</width>
      <height>71</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QSpinBox" name="clr_lg">
    <property name="geometry">
     <rect>
      <x>1330</x>
      <y>580</y>
      <width>70</width>
      <height>30</height>
     </rect>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::UpDownArrows</enum>
    </property>
    <property name="keyboardTracking">
     <bool>true</bool>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="singleStep">
     <number>3</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="clr_lr">
    <property name="geometry">
     <rect>
      <x>1240</x>
      <y>580</y>
      <width>70</width>
      <height>30</height>
     </rect>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::UpDownArrows</enum>
    </property>
    <property name="keyboardTracking">
     <bool>true</bool>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="singleStep">
     <number>3</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="clr_lb">
    <property name="geometry">
     <rect>
      <x>1420</x>
      <y>580</y>
      <width>70</width>
      <height>30</height>
     </rect>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::UpDownArrows</enum>
    </property>
    <property name="keyboardTracking">
     <bool>true</bool>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="singleStep">
     <number>3</number>
    </property>
   </widget>
   <widget class="QRadioButton" name="spline">
    <property name="geometry">
     <rect>
      <x>1250</x>
      <y>650</y>
      <width>99</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>sploshnaya</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">GroupLine</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="stline">
    <property name="geometry">
     <rect>
      <x>1380</x>
      <y>650</y>
      <width>99</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>punktirnaya</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">GroupLine</string>
    </attribute>
   </widget>
   <widget class="QDoubleSpinBox" name="width_l">
    <property name="geometry">
     <rect>
      <x>1380</x>
      <y>690</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::UpDownArrows</enum>
    </property>
    <property name="minimum">
     <double>0.010000000000000</double>
    </property>
    <property name="value">
     <double>0.500000000000000</double>
    </property>
   </widget>
   <widget class="QRadioButton" name="otsut">
    <property name="geometry">
     <rect>
      <x>1230</x>
      <y>890</y>
      <width>99</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>otsutstvuet</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">GroupVertex</string>
    </attribute>
   </widget>
   <widget class="QSpinBox" name="clr_vb">
    <property name="geometry">
     <rect>
      <x>1420</x>
      <y>820</y>
      <width>70</width>
      <height>30</height>
     </rect>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::UpDownArrows</enum>
    </property>
    <property name="keyboardTracking">
     <bool>true</bool>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="singleStep">
     <number>3</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="clr_vg">
    <property name="geometry">
     <rect>
      <x>1330</x>
      <y>820</y>
      <width>70</width>
      <height>30</height>
     </rect>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::UpDownArrows</enum>
    </property>
    <property name="keyboardTracking">
     <bool>true</bool>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="singleStep">
     <number>3</number>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="sz_v">
    <property name="geometry">
     <rect>
      <x>1380</x>
      <y>930</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::UpDownArrows</enum>
    </property>
   </widget>
   <widget class="QSpinBox" name="clr_vr">
    <property name="geometry">
     <rect>
      <x>1240</x>
      <y>820</y>
      <width>70</width>
      <height>30</height>
     </rect>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::UpDownArrows</enum>
    </property>
    <property name="keyboardTracking">
     <bool>true</bool>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="singleStep">
     <number>3</number>
    </property>
   </widget>
   <widget class="QRadioButton" name="krug">
    <property name="geometry">
     <rect>
      <x>1340</x>
      <y>890</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>krug</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">GroupVertex</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="kvadrat">
    <property name="geometry">
     <rect>
      <x>1410</x>
      <y>890</y>
      <width>99</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>kvadrat</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">GroupVertex</string>
    </attribute>
   </widget>
   <widget class="QSpinBox" name="bkground_b">
    <property name="geometry">
     <rect>
      <x>1420</x>
      <y>1070</y>
      <width>70</width>
      <height>30</height>
     </rect>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::UpDownArrows</enum>
    </property>
    <property name="keyboardTracking">
     <bool>true</bool>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="singleStep">
     <number>3</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="bkground_g">
    <property name="geometry">
     <rect>
      <x>1330</x>
      <y>1070</y>
      <width>70</width>
      <height>30</height>
     </rect>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::UpDownArrows</enum>
    </property>
    <property name="keyboardTracking">
     <bool>true</bool>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="singleStep">
     <number>3</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="bkground_r">
    <property name="geometry">
     <rect>
      <x>1240</x>
      <y>1070</y>
      <width>70</width>
      <height>30</height>
     </rect>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::UpDownArrows</enum>
    </property>
    <property name="keyboardTracking">
     <bool>true</bool>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="singleStep">
     <number>3</number>
    </property>
   </widget>
   <widget class="QRadioButton" name="parall">
    <property name="geometry">
     <rect>
      <x>1240</x>
      <y>450</y>
      <width>99</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>parallelnaya</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">GroupProjection</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="central">
    <property name="geometry">
     <rect>
      <x>1360</x>
      <y>450</y>
      <width>99</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>centralnaya</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">GroupProjection</string>
    </attribute>
   </widget>
   <widget class="QPushButton" name="save_screen">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>1230</y>
      <width>131</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>save screenshot</string>
    </property>
   </widget>
   <widget class="QPushButton" name="cr_gif">
    <property name="geometry">
     <rect>
      <x>840</x>
      <y>1230</y>
      <width>131</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>create gif</string>
    </property>
   </widget>
   <widget class="QPushButton" name="stop">
    <property name="geometry">
     <rect>
      <x>1010</x>
      <y>1230</y>
      <width>131</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>stop</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QOpenGLWidget" name="widget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>1200</width>
      <height>1200</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit">
    <property name="geometry">
     <rect>
      <x>1280</x>
      <y>190</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>scale</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_2">
    <property name="geometry">
     <rect>
      <x>1300</x>
      <y>140</y>
      <width>113</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>rotate X Y Z</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_3">
    <property name="geometry">
     <rect>
      <x>1300</x>
      <y>60</y>
      <width>113</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>shift X Y Z</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_4">
    <property name="geometry">
     <rect>
      <x>1280</x>
      <y>620</y>
      <width>161</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>line color(R G B)</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_5">
    <property name="geometry">
     <rect>
      <x>1270</x>
      <y>690</y>
      <width>113</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>line width</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_6">
    <property name="geometry">
     <rect>
      <x>1270</x>
      <y>860</y>
      <width>181</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>vertex color(R G B)</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_7">
    <property name="geometry">
     <rect>
      <x>1280</x>
      <y>930</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>vertex size</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_8">
    <property name="geometry">
     <rect>
      <x>1270</x>
      <y>1110</y>
      <width>181</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>background (R G B)</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="reset">
    <property name="geometry">
     <rect>
      <x>1300</x>
      <y>1150</y>
      <width>131</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>reset model</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="settings">
    <property name="geometry">
     <rect>
      <x>1300</x>
      <y>1220</y>
      <width>131</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>reset settings</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="GroupVertex"/>
  <buttongroup name="GroupLine"/>
  <buttongroup name="GroupProjection"/>
 </buttongroups>
</ui>
